x-pack/plugins/security_solution/public/common/containers/source/use_data_view.tsx

```diff
@@ -51,13 +59,29 @@ export const useDataView = (): { indexFieldsSearch: (selectedDataViewId: string)
     [dispatch]
   );
   const indexFieldsSearch = useCallback(
-    (selectedDataViewId: string) => {
+    (
+      selectedDataViewId: string,
+      scopeId: SourcererScopeName = SourcererScopeName.default,
+      needToBeInit: boolean = false
+    ) => {
       const asyncSearch = async () => {
         abortCtrl.current = {
           ...abortCtrl.current,
           [selectedDataViewId]: new AbortController(),
         };
         setLoading({ id: selectedDataViewId, loading: true });
+        if (needToBeInit) {
+          const dataViewToUpdate = await getSourcererDataview(
```

Comment:
what about the Create Field button in the field browser? after the field is created, we recall `indexFieldsSearch` to refresh the fields. This is in `create_field_button/index.tsx`

Reply:
it should be ok because the data view is already selected. This is needed because we need to fetch the data view missing in the kibana data view.